### PR TITLE
File structure description

### PR DESCRIPTION
**Pull request by JeromeMartinez**
A lot "Work in Progress" documentation, with several things not yet implemented (hashes, handling of several WAV files for one track i.e. for reels, error correcting codes...)

[I update it online](https://docs.google.com/document/d/1eAE_5YkGQ7GEuBsM2x1qOnXgbd6nJ7wDrvEl-rU1md4/edit?usp=sharing) in case one want to help.

### CODE REVIEW

File: Doc/File_Stucture.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
+# Structure of RAWcooked reversibility data 
+
+Beside video and audio content, losslessly compressed with adapted lossless format, some sidecar data is needed for guaranting the reversability to the orignal files.
```
Comment (retokromer) [inline Doc/File_Stucture.md]: for guaranteeing the reversibility to the original

File: Doc/File_Stucture.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
+# Structure of RAWcooked reversibility data 
+
+Beside video and audio content, losslessly compressed with adapted lossless format, some sidecar data is needed for guaranting the reversability to the orignal files.  
+RAWcooked reversibility data is built for a specific file, with specific options. I most cases, RAWcooked reversibility data is still usable after a remux (e.g. from MKV to MOV with RAWcooked reversibility data in a sidecar file) or transcode (e.g. FFV1v1 to FFV1v3, keeping the same tracks in the same order) but it is not guaranted.
```
Comment (retokromer) [inline Doc/File_Stucture.md]: is not guaranteed

File: Doc/File_Stucture.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+File name of the resulting file.  
+
+This element is a marker, indicating that any element after this element and before the next FileName element are releted to this file.  
+
+#### DataSize
+
+Size of the data in the file. The total file size is BeforeData content size + DataSize + AfterData content size.  
+
+If not present, total file size is considred as unknown.
```
Comment (retokromer) [inline Doc/File_Stucture.md]: considered

File: Doc/File_Stucture.md [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+If not present, total file size is considred as unknown.  
+DataSize must be present for all corresponding FileName element in case there are several FileName elements. This is useful for knowing when the tool need to switch to the next file during demux.
+
+Type: unsigned integer
+
+#### BeforeData
+
+Metadata of the the resulting file before the actual content.  
+
+If not present, BeforeData content size is considred as 0.
```
Comment (retokromer) [inline Doc/File_Stucture.md]: considered

File: Doc/File_Stucture.md [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+
+If not present, BeforeData content size is considred as 0.  
+
+Type: Compressed
+
+#### AfterData
+
+Metadata of the the resulting file after the actual content.
+
+If not present, AfterData content size is considred as 0.
```
Comment (retokromer) [inline Doc/File_Stucture.md]: considered

File: Doc/File_Stucture.md [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+#### MaskAdditionAfterData
+
+This element has the same meaning as "BeforeData" element after its content is summed byte per byte to the "MaskBaseAfterData" element content.
+
+Type: Compressed
+
+## Storage of sidecar data
+
+As a general rule, we try to fill some information at a place able to receive some description.  
+The description should contain the text "rawcooked_reversibility_data" or any variant (any letter as uppercase, a space instead of an underscore). The prefered form, when there is no constraint about any legacy systems, is "RAWcooked reversibility data".
```
Comment (retokromer) [inline Doc/File_Stucture.md]: preferred form

**Comment by retokromer** [comment]
Can this be pushed? That way we can work on it as well on GitHub. Thank you!

**Comment by JeromeMartinez** [comment]
I was thinking to a first version with shared edition for people willing to help early (link in first comment) as there may be big changes, you prefer modifications after a merge?

**Comment by JeromeMartinez** [comment]
No

**Comment by retokromer** [comment]
In this case, I would suggest to fix the typos and to push.

**Pull request merged**